lib/Runtime/Base/FunctionBody.h

```diff
@@ -316,13 +316,28 @@ namespace Js
     public:
         // These are public because we don't manage them nor their consistency;
         // the user of this class does.
-        void * address;
+        union address_type_ {
+            address_type_(void *ptr):asPtr(ptr) {}
+            address_type_(Js::JavascriptMethod method):asJsMethod(method) {}
 
-        ProxyEntryPointInfo(void* address, ThreadContext* context = nullptr):
+            void *asPtr;
```

Comment:
> because it will read a field that wasn't directly initialized. Am I wrong?

Perfectly yes.
AFAIK, prior to C++11 all the members of union are initialized to their default constructors. This is no longer an issue.

REF: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2008/n2544.pdf 

IMO, what happens if they were initialized is a better question to ask. 

AFAIK; it is illegal most because the alignment issue. However, the usage here is most likely an `aliasing`..